### DISCUSSION TITLE
Error page 404 - Empty Layout Reference issue

### DISCUSSION THREAD

**Discussion by debuger001** [quasarframework/quasar#15252] [General - Components / Directives / etc]
Hi,
I am unable to figure out this issue on error page of 404. Router is on default setting but quasar dev unable to render error page instead keep hitting this error in console that "QPage.js?9989:20 QPage needs to be a deep child of QLayout". In MainLayout there is QLayout structured and other pages are routing just fine. Please help to solve this issue, I cannot find any solution for it
![Quasar 404 bug - 404page](https://user-images.githubusercontent.com/62402545/211889021-9b06d74c-bb43-4f64-b3e8-38690cb65c65.jpg)
![Quasar 404 bug - router](https://user-images.githubusercontent.com/62402545/211889033-8fbb30ac-f763-4b6f-9191-72b7fed0e68c.jpg)
![Quasar 404 bug (2)](https://user-images.githubusercontent.com/62402545/211889037-8e1a0d19-8755-4fc5-9115-eb06868a6127.jpg)
![Quasar 404 bug](https://user-images.githubusercontent.com/62402545/211889040-08518ec6-0dda-463a-a626-9db82dd6dc7c.jpg)

**Comment by debuger001** [comment]
cc @rstoenescu Please take a look. Thanks

**Reply by pdanpdan** [reply]
Impressive, you can mention the main developer to help you read your code that has e few rows :)
You are using qpage without it being a child of qlayout

**Reply by debuger001** [reply]
Hi Popescu. Yes I wasn't sure who to tag 😂 and its not about lines of code actually but the bug straight after Quasar newly created application. As I mention pages and router code is default added with Quasar create. MainLayout have QLayout and same router-view is working for other pages and error page where only error page is throwing this error. Only happening in newly created projects. Old projects are working fine. If you can pin point the solution that would be helpful. Thanks

Main layout for your reference.
![Quasar 404 bug - MainLayout](https://user-images.githubusercontent.com/62402545/211910486-68b5f0a1-5365-4969-81b2-e871946a93c4.jpg)

**Reply by pdanpdan** [reply]
The template in the error page does not have QPage by default - that was added by you
But you have not moved it to have a parent route with a QLayout

**Reply by debuger001** [reply]
Problem solved with this below update to 404 route.
`  {
    path: '/:catchAll(.*)*',
    component: () => import('layouts/MainLayout.vue'),
    children: [
      { path: '', component: () => import('pages/Error404.vue') }
    ]
  }`
  
  Thanks for your time. Cheers!!
